crates/telio-wg/src/link_detection.rs

```diff
@@ -240,77 +213,167 @@ pub struct LinkDetectionUpdateResult {
     pub link_state: Option<LinkState>,
 }
 
-#[derive(Clone, Copy, Debug)]
-struct NodeInfo {
-    last_tx_timestamp: Instant,
-    last_rx_timestamp: Instant,
-    last_tx_bytes: u64,
-    last_rx_bytes: u64,
-    last_link_state: LinkState,
+enum StateVariant {
+    Down,
+    PossibleDown { count: usize },
```

Comment:
What does `count` refer to in the `PossibleDown` variant?

Reply:
Do you think I should add a diagram? As this one https://github.com/NordSecurity/libtelio/blob/main/crates/telio-traversal/src/endpoint_providers/stun.rs#L383

Reply:
Diagram is nice and it _might_ be beneficial but it might get outdated pretty quickly if the code changes. I suggest on improving the `count` parameter. The issue I am having with it is that it's hard to deduce how long does the `count` work for. As we now have context we know it's most probably seconds as we update once per second. For this I would suggest adding a timestamp in the future when it will be expired and set it to `now + 3 seconds` or something like that :)

By the way, I am not dissing the diagram, I think it's a great tool so if you see a quick way of doing so, it would be nice to have.